### PR TITLE
adds support for x86_64-apple, aarch64-apple, x86_64_linux, aarch64-linux

### PR DESCRIPTION
**Pull request by shahadarsh**
[Here](https://github.com/tembo-io/tembo/releases/tag/0.0.28) is an example release. It's set to pre-release as its only for testing. Release notes are not working. I think I know the reason and will fix it in the next PR.

Please ignore the number of commits. :)

### CODE REVIEW

File: .github/workflows/tembo_release.yml [hunk 1]
```diff
@@ -12,26 +12,70 @@ jobs:
...
       - name: Check out the repo
         uses: actions/checkout@v4
 
+      - name: Install Deps Ubuntu
+        id: install_deps_ubuntu
+        run: |
+          set -x
+          apt update -y
+          apt install pkg-config libssl-dev -y
+        if: matrix.os == 'ubuntu-20.04'
```
Comment (theory) [inline .github/workflows/tembo_release.yml:46]: I suggest using `if: runner.os == 'Linux'` instead, so it should work without modification if you change the vision of Ubuntu.

File: .github/workflows/tembo_release.yml [hunk 2]
```diff
@@ -12,26 +12,70 @@ jobs:
...
+      # TODO: Figure out a better way to fetch version for mac & linux
+      - name: CLI version Mac
+        working-directory: ./tembo-cli
+        id: cli_version_mac
+        run: |
+          set -x
+          cargo install cargo-get
+          TEMBO_CLI_NAME=$(cargo-get package.name)-$(cargo-get package.version)
+          echo "TEMBO_CLI_NAME=$TEMBO_CLI_NAME" >> $GITHUB_ENV
+        if: matrix.os == 'macos-latest'
```
Comment (theory) [inline .github/workflows/tembo_release.yml:66]:
```suggestion
        if: runner.os == 'macOS'
```

File: .github/workflows/tembo_release.yml [hunk 3]
```diff
@@ -12,26 +12,70 @@ jobs:
...
+      - name: CLI version Ubuntu
+        working-directory: ./tembo-cli
+        id: cli_version_ubuntu
+        run: |
+          set -x
+          PKG_VERSION=$(awk -F ' = ' '$$1 ~ /version/ { gsub(/[\\"]/, "", $$2); printf("%s",$$2); exit; }' Cargo.toml)
+          TEMBO_VERSION=${PKG_VERSION#"version = "}
+          TEMBO_CLI_NAME=tembo-cli-${TEMBO_VERSION}
+          echo "TEMBO_CLI_NAME=$TEMBO_CLI_NAME" >> $GITHUB_ENV
+        if: matrix.os == 'ubuntu-20.04'
```
Comment (theory) [inline .github/workflows/tembo_release.yml:77]:
```suggestion
        if: runner.os == 'Linux'
```

**Review by theory — APPROVED** [review]
Can't say much about the build changes, but kibitzed a bit about using forward-compatible conditionals in your workflow. :-)

File: .github/workflows/tembo_release.yml (hunk 1 above)
Comment (shahadarsh) [reply .github/workflows/tembo_release.yml:46]: Yup, good idea. I will change that in the next PR.

**Pull request merged**